### PR TITLE
Add transaction data to contract result

### LINKED ISSUE #4217: Add transaction data to contract result
### Problem

The `eth_getLogs` JSON-RPC API returns logs with data that the mirror node does not include in its response. As a result, the relay has to do additional calls to get the block hash, block number, transaction hash, and transaction number per log.

### Solution

* Add `transaction_hash`, `transaction_index`, and `transaction_result` to `contract_result`
* In `ContractResultServiceImpl`, populate the new fields with information from `Transaction`

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Description**:

* Add `transaction_hash`, `transaction_index`, and `transaction_result` to `contract_result`
* Update REST API database doc

**Related issue(s)**:

Fixes #4217

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by edwin-greene — APPROVED** [review]
Looks good!

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/ContractCreateTransactionHandler.java [hunk 1]
```diff
@@ -197,7 +193,8 @@ private void updateChildFromParent(Contract contract, RecordItem recordItem) {
         switch (type) {
             case CONTRACTCREATEINSTANCE -> updateChildFromContractCreateParent(contract, parentRecordItem);
             case ETHEREUMTRANSACTION -> updateChildFromEthereumTransactionParent(contract, parentRecordItem);
-            default -> {}
+            default -> {
```
Comment (xin-hedera) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/ContractCreateTransactionHandler.java:196]: sonar will complain code smell. no need to fix, can tune IDEA to disable auto placing brace on a new line.

File: hedera-mirror-importer/src/main/resources/db/migration/v1/V1.65.4__contract_result_hash.sql [hunk 2]
```diff
@@ -0,0 +1,24 @@
+alter table if exists contract_result
+    add column if not exists transaction_hash   bytea    null,
+    add column if not exists transaction_index  integer  null,
+    add column if not exists transaction_result smallint null;
+
+with results as (
+    select t.consensus_timestamp,
+           coalesce(et.hash, substring(t.transaction_hash from 1 for 32)) as transaction_hash,
```
Comment (xin-hedera) [inline hedera-mirror-importer/src/main/resources/db/migration/v1/V1.65.4__contract_result_hash.sql:8]:
```suggestion
           coalesce(et.hash, substring(t.transaction_hash from 0 for 32)) as transaction_hash,
```

File: hedera-mirror-rest/.husky/pre-commit [hunk 3]
```diff
@@ -1,9 +1,11 @@
 #!/bin/sh
+set -e
 cd hedera-mirror-rest
+
 if [ -x node/npx ]
 then
-  node/npx --no-install pretty-quick --staged
-else
-  npx --no-install pretty-quick --staged
+  export PATH="${PATH}:node/"
```
Comment (xin-hedera) [inline hedera-mirror-rest/.husky/pre-commit:7]: nit: we can unconditionally prepend `hederra-mirror-rest/node` to `PATH`

**Review by xin-hedera — CHANGES_REQUESTED** [review]

File: hedera-mirror-importer/src/main/resources/db/migration/v1/V1.65.4__contract_result_hash.sql (hunk 2 above)
Comment (xin-hedera) [reply hedera-mirror-importer/src/main/resources/db/migration/v1/V1.65.4__contract_result_hash.sql:8]: oops, I'm confused about whether the `from` starts from 0 or 1.

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**